ballerina/natives.bal

```diff
@@ -67,17 +112,19 @@ public isolated function replaceAll(string originalString, string regex, string
 # + replacement - The replacement string to replace the first substring, which
 #                 matches the regex
 # + return - The resultant string with the replaced substring
+# # Deprecated
+# This function will be removed in a later. Use `replace` instead.
+@deprecated
 public isolated function replaceFirst(string originalString, string regex, string replacement) returns string {
-    handle value = replaceFirstExternal(java:fromString(originalString), java:fromString(regex),
-                                        java:fromString(replacement));
-    string? updatedString = java:toString(value);
-    if updatedString is string {
-        return updatedString;
-    } else {
-        // Should never reach here.
-        error e = error(string `error occurred while replacing ${regex} in ${originalString}`);
-        panic e;
+    handle|error value = trap replaceFirstExternal(java:fromString(originalString), java:fromString(regex),
+                                                   java:fromString(replacement));
+    if value is handle {
+        string? updatedString = java:toString(value);
+        if updatedString is string {
+            return updatedString;
+        }
     }
```

Comment:
What are we going to do when there is an error? IMO we need to panic with proper error message

Reply:
Returns original string when there is an error.

Reply:
Updated in [01395ba](https://github.com/ballerina-platform/module-ballerina-regex/pull/377/commits/01395ba5facc9c84e47af405a9741f23d107a496).